### PR TITLE
chore: trim summary before checking if it's empty

### PR DESCRIPTION
**Pull request by st3iny**
* Resolves: https://github.com/nextcloud/server/pull/45309#discussion_r1600178496

## Summary

As reviewed by Thomas.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by ChristophWurst — APPROVED** [review]

File: apps/dav/lib/CalDAV/CalendarObject.php [hunk 1]
```diff
@@ -104,7 +104,7 @@ private function createConfidentialObject(Component\VCalendar $vObject): void {
 		});
 
 		foreach ($vElements as $vElement) {
-			if (empty($vElement->select('SUMMARY'))) {
+			if (empty(trim($vElement->select('SUMMARY')))) {
```
Comment (tcitworld) [inline apps/dav/lib/CalDAV/CalendarObject.php:107]: I'm thinking about this only now, but `$vElement->select(` actually returns an array, so my comment wasn't a good idea after all. Sorry! :s

**Review by tcitworld — CHANGES_REQUESTED** [review]

File: apps/dav/lib/CalDAV/CalendarObject.php (hunk 1 above)
Comment (st3iny) [reply apps/dav/lib/CalDAV/CalendarObject.php:107]: Ah right. It is an array of properties so let's keep it as is for now.

**Pull request closed without merging**